src/test/java/org/medicmobile/webapp/mobile/OpenSettingsDialogFragmentTest.java

```diff
@@ -0,0 +1,193 @@
+package org.medicmobile.webapp.mobile;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.View.OnTouchListener;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockSettings;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk=28)
+public class OpenSettingsDialogFragmentTest {
```

Comment:
Great tests! Nice to be checking all the different cases. We really need to upgrade to jUnit 5 soon so that we can leverage [parameterized tests](https://www.baeldung.com/parameterized-tests-junit-5) in some of these cases where the test code is basically exactly the same....

Reply:
Nice! I've created [an issue](https://github.com/medic/cht-android/issues/260) to not forget.